### PR TITLE
Jeff iteration2 placing ships

### PR DESCRIPTION
**Pull request by Jredish11**
add tests and methods for placing ships on board, and assigning empty cell objects ship objects

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -34,6 +34,20 @@ def valid_placement?(ship, coordinates)
       false
     end
   end
+
+  def place(ship, coordinates)
+    if valid_placement?(ship, coordinates) #moved this up, if valid_placement?(ship, coordinates) then allow rest of method to run
```
Comment (Jredish11) [inline lib/board.rb:39]: refactored here to continue our use of other  local methods to start off our method definition for place(ship, coordinates)

File: lib/board.rb [hunk 2]
```diff
@@ -34,6 +34,20 @@ def valid_placement?(ship, coordinates)
...
+       coordinates.each do |coordinate|#each method is being called on the coordinates array(from cell class)
+         @cell = @cells[coordinate] #creating local variable for the cell object(@cells[coordinate])
```
Comment (Jredish11) [inline lib/board.rb:41]: created local variable for readability

File: lib/board.rb [hunk 3]
```diff
@@ -34,6 +34,20 @@ def valid_placement?(ship, coordinates)
...
+        @cell.place_ship(ship) #call place_ship method(from cell class) on variable we created for cell object, pass in ship argument
```
Comment (Jredish11) [inline lib/board.rb:42]: place_ship method came from cell class. took forever to realize you can't pull arugment name from another class with the method name.  finally got to pass using ship as argument

File: spec/board_spec.rb [hunk 4]
```diff
@@ -58,4 +58,34 @@
...
   end
+
+  describe '#place' do
+    it 'can place a ship on the board' do
+      @board.place(@cruiser, ['A1', 'A2', 'A3'])
+      @cell_1 = @board.cells["A1"]
+      @cell_2 = @board.cells["A2"]
+      @cell_3 = @board.cells["A3"]
+      expect(@cell_1.ship).to eq(@cruiser)
+      expect(@cell_2.ship).to eq(@cruiser)
```
Comment (Jredish11) [inline spec/board_spec.rb:69]: tests to make sure that ship type in the cell object is a cruiser

File: spec/board_spec.rb [hunk 5]
```diff
@@ -58,4 +58,34 @@
...
+      expect(@cell_3.ship).to eq(@cruiser)
+      require 'pry'; binding.pry   ## hitting a pry @board.cells here i can now see the ship in cells above. need to reach through the cells object to test that both cells have same ship object
```
Comment (Jredish11) [inline spec/board_spec.rb:71]: forgot to delete all this and the #md  .

File: spec/board_spec.rb [hunk 6]
```diff
@@ -58,4 +58,34 @@
...
+      expect(@board.cells["A1"].ship).to eq(@cruiser)
+      expect(@board.cells["A2"].ship).to eq(@cruiser)
+      expect(@board.cells["A3"].ship).to eq(@cruiser)
+      expect(@board.cells["A4"].ship).to eq(nil)
```
Comment (Jredish11) [inline spec/board_spec.rb:75]: making sure that the 'A4' cell object was still nil, no ship inside.

File: spec/board_spec.rb [hunk 7]
```diff
@@ -58,4 +58,34 @@
...
+      expect(@cell_3.ship).to eq(@cell_2.ship)
```
Comment (Jredish11) [inline spec/board_spec.rb:76]: check to see that both cells are currently holding same ship object

File: spec/board_spec.rb [hunk 8]
```diff
@@ -58,4 +58,34 @@
...
+    
+    it 'places another ship on the board' do
+      @board.place(@submarine, ['B1', 'C1'])
+      @cell_1 = @board.cells["B1"]
+      @cell_2 = @board.cells["C1"]
+      expect(@cell_1.ship).to eq(@submarine)
+      expect(@cell_2.ship).to eq(@submarine)
+      expect(@board.cells["B1"].ship).to eq(@submarine)
+      expect(@board.cells["C1"].ship).to eq(@submarine)
+      expect(@board.cells["A1"].ship).to eq(nil)
```
Comment (Jredish11) [inline spec/board_spec.rb:87]: checking to make sure sub is only taking up 2 spaces

File: spec/board_spec.rb [hunk 9]
```diff
@@ -58,4 +58,34 @@
...
+      expect(@cell_1.ship).to eq(@cell_2.ship)
```
Comment (Jredish11) [inline spec/board_spec.rb:88]: check that both cells hold same ship object: submarine

File: lib/board.rb [hunk 10]
```diff
@@ -34,6 +34,20 @@ def valid_placement?(ship, coordinates)
       false
     end
   end
+
+  def place(ship, coordinates)
+    if valid_placement?(ship, coordinates) #moved this up, if valid_placement?(ship, coordinates) then allow rest of method to run
+       coordinates.each do |coordinate|#each method is being called on the coordinates array(from cell class)
```
Comment (Jredish11) [inline lib/board.rb:40]: that markdown should say coordinates argument. and its not from the cell class.

**Comment by Brianisthebest** [comment]
Way to go!
